src/main/java/com/rultor/agents/ec2/AmazonEC2.java

```diff
@@ -0,0 +1,139 @@
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Amazon EC2 instance on demand.
+ *
+ * @author Marton Horvath (marton.horvath@m323.org)
+ * @version $Id$
+ * @since 2.0
+ */
+@Immutable
+public class AmazonEC2 implements Amazon {
+    /**
+     * AWS credentials.
+     */
+    private final transient AWSCredentials credentials;
+    /**
+     * AWS region.
+     */
+    private final transient String region;
+    /**
+     * EC2 instance type.
+     */
+    private final transient String itype;
+    /**
+     * EC2 instance key pair name.
+     */
+    private final transient String keyname;
+
+    /**
+     * Public ctor.
+     * @param cred Credentials
+     * @param zone AWS region
+     * @param type Instance type
+     * @param key Instance key pair name
+     * @checkstyle ParameterNumberCheck (12 lines)
+     */
+    public AmazonEC2(
+        @NotNull final AWSCredentials cred,
+        @NotNull final String zone,
+        @NotNull final String type,
+        @NotNull final String key) {
+        super();
+        this.credentials = cred;
+        this.region = zone;
+        this.itype = type;
+        this.keyname = key;
+    }
+
+    // @checkstyle MagicNumberCheck (30 lines)
+    @Override
+    public final Instance runOnDemand() throws IOException {
+        Instance instance = null;
+        final AmazonEC2Client client = new AmazonEC2Client(
+            this.credentials
+        );
+        client.setEndpoint(this.region);
+        final RunInstancesRequest request = this.createRequest();
+        final List<Instance> instances = client.runInstances(
+            request
+        ).getReservation().getInstances();
+        if ((instances != null) && !instances.isEmpty()) {
+            instance = instances.get(0);
+        }
+        InstanceState state = instance.getState();
+        if (state.getCode() < 16) {
+            final DescribeInstanceStatusRequest statusreq =
+                new DescribeInstanceStatusRequest();
+            statusreq.withInstanceIds(instance.getInstanceId());
+            while (state.getCode() < 16) {
+                try {
+                    TimeUnit.SECONDS.sleep(Tv.FIFTEEN);
+                } catch (final InterruptedException ex) {
+                    continue;
+                }
+                final DescribeInstanceStatusResult instanceStatus = client
+                    .describeInstanceStatus(statusreq);
+                state = instanceStatus.getInstanceStatuses().get(0)
+                    .getInstanceState();
+            }
+        }
+        return instance;
+    }
+
+    /**
+     * Creates a new {@link RunInstancesRequest}.
+     * @return The {@link RunInstancesRequest} created
+     */
+    private RunInstancesRequest createRequest() {
+        final RunInstancesRequest request = new RunInstancesRequest();
+        request.withKeyName(this.keyname);
```

Comment:
please convert it to *fluent* style, and just return: `new RunInstancesRequest().withKeyName(this.keyname).withMinCount(1).....`